### PR TITLE
feat(examples): Add common README.md

### PR DESCRIPTION
**Pull request by razvand**
The common `README.md` file in the `examples/` directory contains common instruction for setting up, configuring, building, running, using and testing applications stored in subdirectories. Specific instructions are to be handled by individual `README.md` file in each application / example subdirectory.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+These are common instructions for example applications.
+Each application gets their own directory, used for documenting, setting up, configuring, building, running, testing and debugging the application.
+Common information, configuration scripts are collected together in other directories such as `../utils`.
+
+## Instructions for Bincompat Apps
+
+The following are common information for examples of applications running in binary compatibility (i.e. **bincompat**) mode.
+Instructions assume you are running commands in the application example subdirectory.
+
+Application running in binary compatibility mode use [`app-elfloader`](https://github.com/unikraft/app-elfloader) images to load and run native Linux ELFs (*Executable and Linking Format* files).
```
Comment (StefanJum) [inline examples/README.md:12]:
```suggestion
Applications running in binary compatibility mode use [`app-elfloader`](https://github.com/unikraft/app-elfloader) images to load and run native Linux ELFs (*Executable and Linking Format* files).
```

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+The images are generated from using `../library/base`.
```
Comment (StefanJum) [inline examples/README.md:13]:
```suggestion
The images are generated using the setup under `../library/base/`.
```

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+We call them `base` images or `base` kernels.
+
+The images are also stored in the Unikraft registry and can be pulled from there.
+
+### Directory Contents
+
+A typical directory for a bincompat app contains:
+
+* `Kraftfile`: build / run rules, including pulling the `base` image
+* `Dockerfile`: filesystem, including binary and libraries
```
Comment (StefanJum) [inline examples/README.md:23]:
```suggestion
* `Dockerfile`: creates the filesystem, including binary and libraries
```

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+   ```
+
+   All scripts start a local instance of the `base` image with the application running in bincompat mode.
+   The application filesystem (binary executable, libraries, configuration files) are packed inside the `rootfs.cpio` initrd file.
+   There is no Kraft script to run Firecracker, as Kraft doesn't currently support running images with networking support using Firecracker.
+
+1. You can run the `strace` or `debug` kernels with the scripts:
+
+   ```console
+   ./run-qemu-x86_64.sh kernels/base_qemu-x86_64-strace
```
Comment (StefanJum) [inline examples/README.md:111]:
```suggestion
   ./run-qemu-x86_64.sh ../../kernels/base_qemu-x86_64-strace
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Looks good @razvand, few minor comments.

**Comment by razvand** [comment]
This is superseded by #36

**Pull request closed without merging**